File: fix_semicolon_errors.py
#!/usr/bin/env python3
"""
Script to fix semicolon errors in import statements.
"""

import os
import re
import glob

def fix_semicolon_errors(file_path):
    """Fix semicolon errors in import statements."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix import statements that have semicolons in wrong places
        # Pattern: import React from 'react'; -> import React from 'react'
        content = re.sub(r"import\s+React\s+from\s+'react';", "import React from 'react'", content)
        content = re.sub(r"import\s+React\s+from\s+\"react\";", "import React from \"react\"", content)
        
        # Fix other common import patterns
        content = re.sub(r"import\s+(\w+)\s+from\s+['\"][^'\"]+['\"];", r"import \1 from '\2'", content)
        
        # Fix specific patterns that were broken
        content = re.sub(r"import\s+React\s+from\s+'react';", "import React from 'react'", content)
        content = re.sub(r"import\s+React\s+from\s+\"react\";", "import React from \"react\"", content)
        
        # Fix any remaining semicolons after import statements
        content = re.sub(r"import\s+[^;]+;", lambda m: m.group(0).rstrip(';'), content)
        
        # Fix any double semicolons
        content = re.sub(r';;+', ';', content)
        
        # Write back if changes were made
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed semicolon errors in: {file_path}")
            return True
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix semicolon errors."""
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'components/**/*.tsx',
        'components/**/*.ts'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            # Skip .original files
            if '.original' in file_path:
                continue
                
            files_processed += 1
            if fix_semicolon_errors(file_path):
                files_fixed += 1
    
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed semicolon errors in {files_fixed} files")

if __name__ == "__main__":
    main()

File: fix-semicolon-errors.js
import fs from 'fs';
import path from 'path';

function fixSemicolonErrors(content) {
  // Fix semicolons after JSX fragments
  content = content.replace(/<Footer \/>;\s*<\/>;/g, '<Footer />\n    </>');
  content = content.replace(/<\/>;\s*\);/g, '</>\n  );');
  
  // Fix extra semicolons after function declarations
  content = content.replace(/};\s*$/gm, '}');
  
  return content;
}

function processFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const fixedContent = fixSemicolonErrors(content);
    
    if (content !== fixedContent) {
      fs.writeFileSync(filePath, fixedContent, 'utf8');
      console.log(`Fixed: ${filePath}`);
    }
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
  }
}

function processDirectory(dirPath) {
  const items = fs.readdirSync(dirPath);
  
  for (const item of items) {
    const fullPath = path.join(dirPath, item);
    const stat = fs.statSync(fullPath);
    
    if (stat.isDirectory()) {
      processDirectory(fullPath);
    } else if (item.endsWith('.tsx') || item.endsWith('.ts')) {
      processFile(fullPath);
    }
  }
}

// Process the app directory
processDirectory('./app');
console.log('Semicolon error fixes completed');